XYZDrivers/src/main/java/com/xyzdrivers/data/ConfiguredConnectionProvider.java

```diff
@@ -0,0 +1,110 @@
+package com.xyzdrivers.data;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.enterprise.context.RequestScoped;
+
+/**
+ *
+ * @author Antony Wyatt
+ * Provides a database connection.
+ */
+@RequestScoped
+public class ConfiguredConnectionProvider implements ConnectionProvider {
+        
+    private Connection connection;
+    private boolean configured = false;
+    private String url, username, password;
+    
+    /**
+     * Provides a database connection to the caller.
+     * @return A new connection, unless a connection has already been provided and is still open.
+     * @throws SQLException
+     */
+    @Override
+    public Connection getConnection() throws SQLException {
+        
+        if (connection == null || connection.isClosed())
+        {           
+            if (!configured) {
+                readConfiguration();
+            }
+            
+            connection = DriverManager.getConnection(url, username, password);
+        }
+        
+        return connection;
+        
+    }
+    
+    /**
+     * Closes the provided connection if it is still in use.
+     * @throws java.sql.SQLException
+     */
+    @Override
+    public void close() throws SQLException {
+        
+        if (connection != null && !connection.isClosed())
+        {
+            connection.close();
+        }
+        
+    }
+    
+    private Properties readConfigurationProperties(String configurationFileName) {
```

Comment:
Shouldn't this method declare a throws for IlligalArgumentEx?

Reply:
Nope! IllegalArgumentException is unchecked

Reply:
Interesting, didnt know this! Neat